_studio/mfx_lib/encode_hw/mjpeg/src/mfx_mjpeg_encode_hw.cpp

```diff
@@ -972,6 +972,7 @@ mfxStatus MFXVideoENCODEMJPEG_HW::EncodeFrameCheck(
 
     bs->TimeStamp = surface->Data.TimeStamp;
     bs->DecodeTimeStamp = surface->Data.TimeStamp;
+    bs->FrameType = MFX_FRAMETYPE_I;
```

Comment:
What about SW path? And are there a shared SW/HW code where we can do that?

Reply:
SW path is covered by change in mfx_mjpeg_encode.cpp. I found no right place to put `MFX_FRAMETYPE_I` assignment that is called from `EncodeFrameAsync` and shared by HW and SW pathes.